Tiddly space icons update sept

jdlrobson opened the pull request:
enriches modifier information / space name with uris.

jdlrobson committed:
turned modifier/space name into link

this aids discovery and hadn't been done previosuly due to laziness.
A user can now go straight to a users home space linking modifier to person.

jdlrobson committed:
added special handling for shadow and missing icons.

Now the label saying missing only shows if a missingIcon is present. This allows a user
to make use of it (but if the icon is not present unknown modifier is rather confusing -
the absence of the image alone should hint that the modifier is unknown).

Shadow tiddlers can have their own icon shadowIcon however if this is not present it should
be treated as a core tiddler - just like tiddlers in the bag tiddlyspace and system are.

jdlrobson committed:
enforce lowercase modifier names.

In TiddlySpace users have lowercase names. I noticed a few plugins in the system
bag had modifiers such as FND which was throwing 404s on SiteIcons. Prevent this
state from ever occuring.

jdlrobson committed:
increment version number

cdent commented:
There's something not quite right with this. If you create a new space, the left icon on the GettingStarted tiddler (from system-info) has a tooltip of: "from <a href="some path">system-info</a>"

When clicking on the actual link (below the icon), the concertina opens _and_ the browser opens to the other space.

On the right icon (osmosoft) it would be nice if the icon itself was clickable, not just the text below it.

cdent commented:
Note: the quoted tooltip info above, I wanted literal &lt; and &gt; not what github chose to do

jdlrobson commented:
good spot. Will fix now.

jdlrobson committed:
added optional tooltip argument to labelOptions

allows the tooltip to be different to the label text.
(previously if html label was used the tooltip would show escaped html = bad)

jdlrobson commented:
I've fixed the tooltip issue. The reason the Osmosoft logo is not clickable currently is that I was worried that if it appeared clickable a user might think it would have the same function as the private/public/space icon and would cause an interaction with the concertina when instead it would navigate away from the page.

Open for debate though - this is something that would be easy enough to do.

cdent commented:
Well, as I hope you know from other things I've written, I want ALL space icons, everywhere, to be links to their associated space, but we can get a third opinion on that and worry about it some other time.

I'll go ahead and merge this stuff in.

cdent commented:
Looks like maybe you didn't push your code?

jdlrobson commented:
you'd be right! pushed now. No I agree in the principle of all space icons pointing to space, I'm just worrying about this context and confusing people in these early days..
